packages/cyberstorm/src/components/Breadcrumb.tsx

```diff
@@ -0,0 +1,86 @@
+import { Fragment, FunctionComponent, ReactNode } from "react";
+
+import { faHouse, faSlashForward } from "@fortawesome/pro-regular-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import styles from "./componentStyles/Breadcrumb.module.css";
+import { IndexLink } from "./Links";
+
+type Label = ReactNode | string;
+
+interface Crumb {
+  label: Label;
+  LinkComponent?: FunctionComponent;
+}
+
+type LinkCrumb<P> = Omit<Crumb, "LinkComponent"> & {
+  label: Label;
+  LinkComponent: FunctionComponent<P>;
+  LinkProps: P;
+};
+
+interface BreadCrumbProps<P> {
+  parts: (Crumb | LinkCrumb<P>)[];
```

Comment:
I don't think this is what we want. It might work on the storybook stories, and maybe even on some pages, but sooner or later we will have a page that will not work with this.

The problem is that this assumes all passed `LinkCrumbs` share the same generic type `<P>`. The example below would not work, since the two links use different props that are incompatible with each other:

```typescript
export const MyCrumbs = () => {
  const crumbs = [
    {
      LinkComponent: TeamLink,
      LinkProps: { team: "Testruction" },
      label: "Testruction",
    },
    {
      LinkComponent: CommunityLink,
      LinkProps: { community: "ror2" },
      label: "Risk of Rain 2",
    },
  ];
  return <BreadCrumb parts={crumbs} />;
};
```

I consider this a blocker.

Reply:
Latest commit has a suggestion on how this could be solved(?)

Reply:
As mentioned in Discord, I consider typing this (and other component implementations) properly a requirement for them being merged. In case it's not possible to declare the types in a supported way it's of course understandable if they aren't, but as far as I know, it should be doable here. I'd suggest separating the link library implementation to a standalone PR and basing your other (and this) work on that branch to avoid the type declarations of the crumb component from blocking said other work.

Reply:
I consider this comment resolved, but I'll leave it open so I don't hide Mythic's comment above.